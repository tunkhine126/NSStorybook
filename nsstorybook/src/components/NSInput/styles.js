import { makeStyles } from '@material-ui/core/styles';
import colors from 'global-styles';
import styled from 'styled-components';

export const ComponentStyle = styled.div`
  .MuiFormLabel-root {
    color: ${colors.TEXT.light};
    font-weight: normal;
  }
  .MuiFormLabel-root.Mui-disabled {
    color: ${colors.STATE.disabled};
  }
  .MuiSvgIcon-root {
    color: ${colors.INTERFACE.ui6};
  }
`;

export const useNSInput = makeStyles(theme => ({
  root: {
    border: `1px solid ${colors.INTERFACE.ui4}`,
    overflow: 'hidden',
    borderRadius: 2,
    backgroundColor: colors.INTERFACE.ui1,
    transition: theme.transitions.create(['border-color']),
    '&:hover': {
      backgroundColor: colors.INTERFACE.ui1,
      borderColor: theme.palette.primary.main,
    },
    '&$focused': {
      backgroundColor: colors.INTERFACE.ui1,
      border: `2px solid ${theme.palette.primary.main}`,
    },
    '&.Mui-disabled': {
      backgroundColor: 'transparent',
    },
  },
  disabled: {
    color: colors.STATE.disabled,
    '&:hover': {
      borderColor: colors.INTERFACE.ui4,
    },
  },
  focused: {},
}));
