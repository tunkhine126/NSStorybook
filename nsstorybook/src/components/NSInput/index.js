/**
 *
 * NSInput
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';

import { useNSInput, ComponentStyle } from './styles';

function NSInput({ icon: Icon, position, ...rest }) {
  const classes = useNSInput();
  let textField;

  if (Icon) {
    textField = (
      <TextField
        InputProps={{
          classes,
          disableUnderline: true,
          [`${position}Adornment`]: (
            <InputAdornment position={position}>
              <Icon />
            </InputAdornment>
          ),
        }}
        {...rest}
      />
    );
  } else {
    textField = (
      <TextField
        InputProps={{
          classes,
          disableUnderline: true,
        }}
        {...rest}
      />
    );
  }
  return <ComponentStyle>{textField}</ComponentStyle>;
}

NSInput.propTypes = {
  icon: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
    PropTypes.object,
  ]),
  position: PropTypes.oneOf(['start', 'end']),
};

export default NSInput;
