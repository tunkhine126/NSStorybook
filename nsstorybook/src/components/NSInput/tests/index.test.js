/**
 *
 * Tests for NSInput
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import NSInput from '../index';

describe('<NSInput />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<NSInput />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<NSInput />);
    expect(firstChild).toMatchSnapshot();
  });
});
