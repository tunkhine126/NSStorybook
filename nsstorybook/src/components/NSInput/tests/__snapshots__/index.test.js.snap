// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NSInput /> Should render and match the snapshot 1`] = `
<div
  class="styles__ComponentStyle-zscvjs-0 iXrycp"
>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root makeStyles-root-41 MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
